ctx.go

```diff
@@ -1367,6 +1370,81 @@ func (c *Ctx) SendFile(file string, compress ...bool) error {
 	return nil
 }
 
+// SendFileWithConfig transfers the file from the given path.
+// Some config fields are overwritten. You can change them by config parameter.
+// Sets the Content-Type response HTTP header field based on the filenames extension.
+func (c *Ctx) SendFileWithConfig(file string, config ...SendFile) error {
```

Comment:
the code looks very similar to the sendFile method, can we swap the core of both methods into a private and control both with it

or at least that the SendFile method uses SendFileWithConfig 

```go
func (c *Ctx) SendFile(file string, compress ...bool) error {
     return c.SendFileWithConfig(file, SendFile{Compress: compress})
}
```


would be good if we could remove as many duplicate variables and keep the code small

Reply:
Right. I think sendfile should use sendfilewithconfig.

Reply:
done